### PR TITLE
LibGL: Use integer comparison for GL_EQUAL and GL_NOTEQUAL

### PR DESCRIPTION
**Pull request by Quaker762**
This is an interesting quirk that occurs due to us using the x87 FPU
when Serenity is compiled for the i386 target. When we calculate our
depth value to be stored in the buffer, it is an 80-bit x87
floating point number, however, when stored into the DepthBuffer,
this is truncated to 32 bits. This 38 bit loss of precision means
that when x87 `FCOMP` is eventually used here the comparison fails.

This could be solved by using a `long double` for the depth buffer,
however this would take up significantly more space and is completely
overkill for a depth buffer. As such, comparing the first 32-bits of
this depth value is "good enough" that if we get a hit on it being
equal, we can pretty much guarantee that it's actually equal.

### CODE REVIEW

**Comment by Quaker762** [comment]
Before: 
image

After: 
image

File: Userland/Libraries/LibGL/SoftwareRasterizer.cpp [hunk 1]
```diff
@@ -278,10 +278,29 @@ static void rasterize_triangle(const RasterizerOptions& options, Gfx::Bitmap& re
                             pass = z >= *depth;
                             break;
                         case GL_NOTEQUAL:
+#ifdef __SSE__
                             pass = z != *depth;
+#else
+                            pass = __builtin_bit_cast(u32, z) != __builtin_bit_cast(u32, *depth);
```
Comment (alimpfard) [inline Userland/Libraries/LibGL/SoftwareRasterizer.cpp:284]: Instead, let's redefine `AK/BitCast.h`'s version of `bit_cast` in terms of this intrinsic, please also guard it with `#if __has_builtin(__builtin_bit_cast)` :)

Comment (Hendiadyoin1) [reply Userland/Libraries/LibGL/SoftwareRasterizer.cpp:284]: I proposed that, with the thinking that our bit_cast might not always produce the most optimal code the compiler can, due to usign a __builtin_memcpy, although after looking at 
https://godbolt.org/z/orWxnTjvn
both seem to be equivalent, my bad....

Comment (alimpfard) [reply Userland/Libraries/LibGL/SoftwareRasterizer.cpp:284]: It's alright to use the intrinsic if the compiler has it, but let's put that ugliness in the implementation of `bit_cast()`.

Comment (Quaker762) [reply Userland/Libraries/LibGL/SoftwareRasterizer.cpp:284]: > Instead, let's redefine `AK/BitCast.h`'s version of `bit_cast` in terms of this intrinsic, please also guard it with `#if __has_builtin(__builtin_bit_cast)` :)

Sure!

**Pull request merged**